extensions/wikia/Email/SpecialSendEmailController.class.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Email;
+
+
+class SpecialSendEmailController extends \WikiaSpecialPageController {
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const REQUIRED_PERMISSION = "staff";
+
+	public function __construct() {
+		parent::__construct( 'SendEmail', '', $listed = false );
+	}
+
+	/**
+	 * Run before index. First make sure the user can access page, then set the title and
+	 * add the css.
+	 */
+	public function init() {
+		$this->assertCanAccess();
+		$this->setTitle();
+		$this->addCss();
+	}
+
+	/**
+	 * Makes sure the current user is staff and can therefore access this page. If not, display
+	 * a permission error page.
+	 * @throws \PermissionsError
+	 */
+	private function assertCanAccess() {
+		if ( !$this->wg->User->isStaff() ) {
+			throw new \PermissionsError( self::REQUIRED_PERMISSION );
```

Comment:
So ```staff``` is actually a group rather than a permission.  Users belong to groups, groups have permission to do certain actions.  The ```PermissionsError``` constructor wants the action that was attempted, something like 'edit' or 'update' etc and constructs a message string from it like ```action-edit``` to use in the error response.

There is actually a ```sendemail``` action defined that you could use for the purpose of this error.  If you wanted to really be by the books, you'd create your own action for this case (e.g. ```testemail```), set it up so that the ```staff``` group has permission to do it, then do ```$user->can( 'testemail' )```.  Could pass ```testemail``` to ```PermissionsError```, and then add a i18n key ```action-testemail```.

Or you could just reuse ```sendemail``` ;)